auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -645,6 +666,28 @@ public IdpConfig build() {
             public FacebookBuilder() {
                 //noinspection deprecation taking a hit for the backcompat team
                 super(FacebookAuthProvider.PROVIDER_ID);
+
+                try {
+                    //noinspection unused to possibly throw
+                    Class c = com.facebook.FacebookSdk.class;
+                } catch (NoClassDefFoundError e) {
+                    throw new RuntimeException(
+                            "Facebook provider cannot be configured " +
+                                    "without dependency. Did you forget to add " +
+                                    "'com.facebook.android:facebook-login:VERSION' dependency?");
+                }
+
+                if (getApplicationContext().getString(R.string.facebook_application_id)
+                        .equals(UNCONFIGURED_CONFIG_VALUE)) {
+                    throw new RuntimeException("Facebook provider unconfigured. Make sure to add" +
+                            " a `facebook_application_id` string. See the docs for more info:" +
+                            " https://github.com/firebase/FirebaseUI-Android/blob/master/auth/README.md#facebook");
+                }
+                if (getApplicationContext().getString(R.string.facebook_login_protocol_scheme)
+                        .equals("fbYOUR_APP_ID")) {
+                    Log.w("FacebookBuilder", "Facebook provider unconfigured for Chrome" +
```

Comment:
is there any purpose for this hardcoded `Log.w`?

Reply:
Laziness? Fixed.